armory/instrument/config.py

```diff
@@ -0,0 +1,313 @@
+def total_wer(sample_wers):
+    """
+    Aggregate a list of per-sample word error rate tuples (edit_distance, words)
+        Return global_wer, (total_edit_distance, total_words)
+    """
+    # checks if all values are tuples from the WER metric
+    if all(isinstance(wer_tuple, tuple) for wer_tuple in sample_wers):
+        total_edit_distance = 0
+        total_words = 0
+        for wer_tuple in sample_wers:
+            total_edit_distance += wer_tuple[0]
+            total_words += wer_tuple[1]
+        if total_words:
+            global_wer = float(total_edit_distance / total_words)
+        else:
+            global_wer = float("nan")
+        return global_wer, (total_edit_distance, total_words)
+    else:
+        raise ValueError("total_wer() only for WER metric aggregation")
+
+
+# TODO: move to armory.utils.metrics
+def identity(*args):
+    return args
+
+
+# TODO: move to armory.utils.metrics
+class MeanAP:
+    def __init__(self, ap_metric):
+        self.ap_metric = ap_metric
+
+    def __call__(self, *args, **kwargs):
+        ap = self.ap_metric(*args, **kwargs)
+        mean_ap = np.fromiter(ap.values(), dtype=float).mean()
+        return {"mean": mean_ap, "class": ap}
+
+
+class ResultsLogWriter(LogWriter):
+    """
+    Logs successful results (designed for task metrics)
+    """
+
+    def __init__(
+        self,
+        adversarial=False,
+        targeted=False,
+        used_preds_as_labels=False,
+        log_level: str = "SUCCESS",
+    ):
+        super().__init__(log_level=log_level)
+        if targeted:
+            if adversarial:
+                self.wrt = "target"
+                self.task_type = "adversarial"
+            else:
+                raise ValueError("benign task cannot be targeted")
+        elif adversarial:
+            if used_preds_as_labels:
+                self.wrt = "benign predictions as"
+            else:
+                self.wrt = "ground truth"
+            self.task_type = "adversarial"
+        else:
+            self.wrt = "ground truth"
+            self.task_type = "benign"
+
+    def _write(self, name, batch, result):
+        if name == "word_error_rate":
+            total, (num, denom) = total_wer(result)
+            f_result = f"total={total:.2%}, {num}/{denom}"
+        elif name in MEAN_AP_METRICS:
+            f_result = f"{result}"
+        elif name in QUANTITY_METRICS:
+            # Don't include % symbol
+            f_result = f"{np.mean(result):.2}"
+        else:
+            f_result = f"{np.mean(result):.2%}"
+        log.success(
+            f"{name} on {self.task_type} examples w.r.t. {self.wrt} labels: {f_result}"
+        )
+
+
+def _task_metric(name, metric_kwargs, use_mean=True, include_target=True):
+    metric = metrics.get_supported_metric(name)
+    final_kwargs = {}
+    if name in MEAN_AP_METRICS:
+        final_suffix = name
+        final = MeanAP(metric)
+        final_kwargs = metric_kwargs
+
+        name = "identity"
```

Comment:
Why is `name` being set to `"identity"`?

Reply:
Yeah, this should be something to indicate it's an input. Changed.